leader/src/cli.rs

```diff
@@ -18,6 +23,63 @@ pub(crate) struct Cli {
     pub(crate) prover_state_config: CliProverStateConfig,
 }
 
+#[derive(Clone, Debug)]
+pub(super) enum BlockNumbers {
+    Single(u64),
+    RangeInclusive(RangeInclusive<u64>),
+}
+
+impl FromStr for BlockNumbers {
+    type Err = anyhow::Error;
+
+    /// Parses input block numbers to determine a specific block or a range of
+    /// blocks.
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::from_str_intern(s)
+            .with_context(|| {
+                format!(
+                    "Expected a single value or a range, but instead got \"{}\".",
+                    s
+                )
+            })
+            .map_err(|e| anyhow!("{e:#}"))
+    }
+}
+
+impl BlockNumbers {
+    fn from_str_intern(s: &str) -> anyhow::Result<Self> {
+        // Did we get passed a single value?
+        if let Ok(v) = s.parse::<u64>() {
+            return Ok(Self::Single(v));
+        }
+
+        // Check if it's a range.
+        let mut range_vals = s.split("..=");
```

Comment:
I think we should either emphasize in the README that the range `has` to be inclusive, otherwise it will fail at runtime, or allow both exclusive and inclusive ranges (and adapt internally the `end`)?